department-of-veterans-affairs/va.gov-cms
# VACMS-7422 vagovclaro homepage tweaks

## Description

Relates to #7422 and other issues surfaced in this airtable: https://airtable.com/applHskzNEaJ4qM7R/tblvm17AmjxYtpwWQ/viw51xsCmfmSgXlOj?blocks=hide 

- Adjusts positioning of pencil icon.
- Removes period from `Login with PIV.`
- Adds template for unauth 403 redirect handling. e.g. if you try to visit `/node/26856/edit` while logged out, the correct login screen will be shown now.
- Adds error handling to the login form. When a user/pass doesn't work, you should see the form input with the error message. Previously, upon validation error the form would rebuild with the error markup visible, but toggled to the PIV login state, so you couldn't see the input with red border & text.

Does not fix the display of the alert on the log in page! Going to tidy alerts up holistically next, there are a few reported issues for them.

## Testing done
visual for pencil/period
logged in through `/node/*/edit` and verified was redirected to the correct form
entered incorrect information in login form and saw error messages instead of the piv button.

## Screenshots
login page visible from unauth `node/*/edit` route:
<img width="676" alt="Screen Shot 2022-01-04 at 3 34 05 PM" src="https://user-images.githubusercontent.com/11279744/148139471-a14c03c4-a5e8-4252-8b29-03535cb0c2c5.png">

input errors now correctly keep the form inputs showing. disregard the ugly alert style still 🙈 
<img width="679" alt="Screen Shot 2022-01-04 at 3 53 04 PM" src="https://user-images.githubusercontent.com/11279744/148139607-9260bddd-a404-4d3d-afbd-cb63735982d7.png">


## QA steps

As unauth user, visit login page and see that pencil is repositioned. The period is gone on the button.

Attempt to login with incorrect user/pass credentials, verify that the login form does not show the PIV button and the input is highlighted correctly.
Fix info and log in correctly.

Go to any node edit form, or admin route, etc, and copy paste the url. Logout, now hit that url again. You should be correctly shown the new login screen as an unauthenticated user. Logging in should display the page with access.

### Definition of Done


### Select Team for PR review


### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
